web-app/js/smartR/_angular/directives/fetchButton.js

```diff
@@ -52,15 +52,15 @@ window.smartRApp.directive('fetchButton', [
                 // we add this conditional $watch because there is some crazy promise resolving for allSamples
                 // going on. This is a workaround which observes allSamples and uses it as criteria for successful
                 // completion.
-                if (scope.showSummaryStats) {
-                    scope.$watch('summaryData', function(newValue) {
-                        // prevents initial firing
-                        if (scope.running && Object.keys(newValue).indexOf('subsets') !== -1) {
-                            scope.allSamples = scope.summaryData.allSamples;
-                            _onSuccess();
-                        }
-                    }, true);
-                }
+                scope.$watch('summaryData', function(newValue) {
+                    if (scope.summaryData &&
+                            scope.showSummaryStats &&
+                            scope.running &&
+                            Object.keys(newValue).indexOf('subsets') !== -1) {
+                        scope.allSamples = scope.summaryData.allSamples;
```

Comment:
why not 

`scope.allSamples = newValue.allSamples;`

so that you are sure it's from the updated summaryData